dace/frontend/python/newast.py

```
@@ -3176,20 +3176,13 @@ def visit_Assign(self, node: ast.Assign):
     def visit_AnnAssign(self, node: ast.AnnAssign):
         type_name = rname(node.annotation)
         try:
-            dtype = eval(type_name)
-            if not isinstance(dtype, dtypes.typeclass):
-                raise NotImplementedError
+            dtype = eval(astutils.unparse(node.annotation), self.globals,
```

We lost here the `dtypes.typeclass` check. The previous code in those lines would extract the `dace.whatever` name, even from an array definition, and check if this is a supported typeclass. Now, it may be possible to annotate an assignment with an unsupported 3rd party type without raising an exception.

---

changing `eval` to something that uses the globals (like N for example) can now correctly resolve type annotations. This caused new issues for array types that would previously silently fail in any case.

How do you propose we solve this?

---

What I mean is that we had two checks, one of typeclass and one evaluating for array types. We lost the typeclass one. I restored it (in a more modern version) and added a test. I am not sure which should be our policy. It will now throw the warning and ignore the annotation, but maybe we want to throw NotImplementedError.